pkg/chunked/tar_split_linux.go

```diff
@@ -0,0 +1,68 @@
+package chunked
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/vbatts/tar-split/archive/tar"
+	"github.com/vbatts/tar-split/tar/storage"
+)
+
+// iterateTarSplit calls handler for each tar header in tarSplit
+func iterateTarSplit(tarSplit []byte, handler func(hdr *tar.Header) error) error {
+	// This, strictly speaking hard-codes undocumented assumptions about how github.com/vbatts/tar-split/tar/asm.NewInputTarStream
+	// forms the tar-split contents. Pragmatically, NewInputTarStream should always produce storage.FileType entries at least
+	// for every non-empty file, which constraints it basically to the output we expect.
+	//
+	// Specifically, we assume:
+	// - There is a separate SegmentType entry for every tar header, but only one SegmentType entry for the full header incl. any extensions
+	// - (There is a FileType entry for every tar header, we ignore it)
+	// - Trailing padding of a file, if any, is included in the next SegmentType entry
+	// - At the end, there may be SegmentType entries just for the terminating zero blocks.
+
+	unpacker := storage.NewJSONUnpacker(bytes.NewReader(tarSplit))
+	for {
+		tsEntry, err := unpacker.Next()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return fmt.Errorf("reading tar-split entries: %w", err)
+		}
+		switch tsEntry.Type {
+		case storage.SegmentType:
+			payload := tsEntry.Payload
+			// This is horrible, but we don’t know how much padding to skip. (It can be computed from the previous hdr.Size for non-sparse
+			// files, but for sparse files that is set to the logical size.)
```

Comment:
This is a pretty horrible hack.

It should be possible for `tar-split/archive/tar` to expose the expected padding size in `Header`.

(Alternatively, the OCI spec suggests that sparse files should not be used, but I don’t expect that can be relied upon.)